src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -0,0 +1,568 @@
+
+
+def delete_continuous_patch_v1(cmd, registry, dryrun):
+    logger.debug("Entering delete_continuous_patch_v1")
+    cssc_tasks_exists, _ = check_continuous_task_exists(cmd, registry)
+    cssc_config_exists = check_continuous_task_config_exists(cmd, registry)
+    if not dryrun and (cssc_tasks_exists or cssc_config_exists):
+        cssc_tasks = ', '.join(CONTINUOUSPATCH_ALL_TASK_NAMES)
+        logger.warning(f"All of these tasks will be deleted: {cssc_tasks}")
+        for taskname in CONTINUOUSPATCH_ALL_TASK_NAMES:
+            # bug: if one of the deletion fails, the others will not be attempted, we need to attempt to delete all of them
+            _delete_task(cmd, registry, taskname, dryrun)
+            logger.warning(f"Task {taskname} deleted.")
+        logger.warning(f"Deleting {CSSC_WORKFLOW_POLICY_REPOSITORY}/{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG}:{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_V1}")
+        delete_oci_artifact_continuous_patch(cmd, registry, dryrun)
+
+    if not cssc_tasks_exists:
+        logger.warning(f"{ERROR_MESSAGE_WORKFLOW_TASKS_DOES_NOT_EXIST}")        
+
+def list_continuous_patch_v1(cmd, registry):
+    logger.debug("Entering list_continuous_patch_v1")
+    cssc_tasks_exists, _ = check_continuous_task_exists(cmd, registry)
+    if not cssc_tasks_exists:
+        logger.warning(f"{ERROR_MESSAGE_WORKFLOW_TASKS_DOES_NOT_EXIST}")
+        return
+
+    acr_task_client = cf_acr_tasks(cmd.cli_ctx)
+    resource_group_name = parse_resource_id(registry.id)[RESOURCE_GROUP]
+    tasks_list = acr_task_client.list(resource_group_name, registry.name)
+    filtered_cssc_tasks = _transform_task_list(tasks_list)
+    return filtered_cssc_tasks
+
+
+def acr_cssc_dry_run(cmd, registry, config_file_path, is_create=True, remove_internal_statements=True):
+    logger.debug(f"Entering acr_cssc_dry_run with parameters: {registry} {config_file_path}")
+    cssc_tasks_exists, _ = check_continuous_task_exists(cmd, registry)
+
+    if config_file_path is None:
+        if not cssc_tasks_exists:
+            raise InvalidArgumentValueError("--config parameter is needed to perform dry-run check.")
+
+        # attempt to get the config file from the registry, since the configuration should exist
+        logger.debug("Retrieving the configuration file from the registry.")
+        _, config_file_path = get_oci_artifact_continuous_patch(cmd, registry)
+        if config_file_path is None:
+            raise AzCLIError("Failed to retrieve the configuration file from the registry.")
+
+    if is_create and cssc_tasks_exists:
+        raise AzCLIError(f"{ERROR_MESSAGE_WORKFLOW_TASKS_ALREADY_EXISTS}")
+
+    file_name = None
+    tmp_folder = None
+    acr_archive_utils_logger_level = acr_archive_utils_logger.getEffectiveLevel()
+    try:
+        file_name = os.path.basename(config_file_path)
+        tmp_folder = os.path.join(os.getcwd(), tempfile.mkdtemp(prefix="cli_temp_cssc"))
```

Comment:
Suggestion: consider using with tempfile.TemporaryDirectory(prefix="cli_temp_cssc") as tmp_folder to generate the temp folder